File: 0841-keys-and-rooms/0841-keys-and-rooms.py
class Solution:
    def canVisitAllRooms(self, rooms: List[List[int]]) -> bool:
        visited={0}
        keys=set()
        for i in rooms[0]:
            keys.add(i)
            visited.add(i)
        while(keys):
            a=keys.pop()
            for i in rooms[a]:
                if i not in keys and i not in visited:
                    keys.add(i)
                    visited.add(i)
        return len(visited)==len(rooms)
            
        
            
        

File: 0841-keys-and-rooms/0841-keys-and-rooms.cpp
class Solution {
public:
    bool canVisitAllRooms(vector<vector<int>>& rooms) {
        vector<bool> visited(rooms.size(),false);
        visited[0]=true;
        vector<int>keys;
        for(auto x:rooms[0])keys.push_back(x);
        while(keys.size()!=0)
        {
         int curr=keys.front();
        keys.erase(keys.begin());
            visited[curr]=true;
            for(auto x: rooms[curr])
            {
                if(visited[x]==false)
                {keys.push_back(x);
                visited[x]=true;}
            }
            
        }
        for(int i=0;i<rooms.size();i++)if(visited[i]==false)return false;
        return true;
    }
};